src/cswinrt/code_writers.h

```diff
@@ -6589,6 +6590,22 @@ private struct InterfaceTag<I>{};
                         w.write("_defaultLazy = new Lazy<%>(() => (%)new SingleInterfaceOptimizedObject(typeof(%), _inner));", default_interface_name, default_interface_name, default_interface_name);
                     }
                 }),
+            [&](writer& w)
+            {
+                if (!gc_pressure_amount) return;
+                w.write("GC.AddMemoryPressure(%);\n", gc_pressure_amount);
+            },
+            [&](writer& w)
+            {
+                if (!gc_pressure_amount) return;
+                w.write(R"(~%()
```

Comment:
Is this adding finalizers for almost all types that didn't have one before this PR? Finalizers have non trivial overhead (especially if this would be for so many objects), are we sure it's something worth adding just to track the memory pressure? IIRC the GC wasn't really doing much at all with this information anyway.

Reply:
Its not all types, it is only a handful which have the special `GCPressure` attribute to indicate their native objects use significant memory.

https://learn.microsoft.com/en-us/uwp/api/windows.foundation.metadata.gcpressureattribute?view=winrt-22621

Reply:
Oh I see, so we're only opting in with a finalizer for those and the others just get that baseline from `IObjectReference` which already has a finalizer anyway, got it. Thank you! 🙂